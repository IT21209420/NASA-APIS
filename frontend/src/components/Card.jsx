import React, { useState } from "react";
import Modal from "react-modal";
import { TransformWrapper, TransformComponent } from "react-zoom-pan-pinch";

Modal.setAppElement("#root");

const Card = ({ photo }) => {
  const [modalIsOpen, setModalIsOpen] = useState(false);

  const openModal = () => {
    setModalIsOpen(true);
  };

  const closeModal = () => {
    setModalIsOpen(false);
  };
  return (
    <div className="border rounded-md p-2 h-52 w-64 border-gray-700  backdrop-filter backdrop-blur-sm">
      <img
        src={photo.img_src}
        alt={photo.camera.full_name}
        className="h-36 w-full object-cover cursor-pointer"
        onClick={openModal}
      />
      <div className="mt-2">
        <h2
          className="text-sm font-semibold
        text-gray-100
        "
        >
          {photo.rover.name}
        </h2>
        <p
          className="text-xs
        text-gray-300
        "
        >
          {photo.earth_date}
        </p>
      </div>

      <Modal
        isOpen={modalIsOpen}
        onRequestClose={closeModal}
        className="flex items-center justify-center h-full"
        shouldCloseOnOverlayClick={true}
        style={{
          overlay: {
            display: "flex",
            alignItems: "center",
            justifyContent: "center",
          },
          content: {
            position: "relative",
            inset: "auto",
          },
        }}
      >
        <button onClick={closeModal}>Close</button>
        <TransformWrapper>
          <TransformComponent>
            <img
              src={photo.img_src}
              alt={photo.camera.full_name}
              style={{
                maxWidth: "90vw",
                maxHeight: "90vh",
                objectFit: "contain",
                zIndex: 1000,
              }}
            />
          </TransformComponent>
        </TransformWrapper>
      </Modal>
    </div>
  );
};

export default Card;
