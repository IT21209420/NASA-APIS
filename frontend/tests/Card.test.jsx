import { describe, it, expect, beforeEach } from "vitest";
import { render } from "@testing-library/react";

import Card from "../src/components/Card";

// import { useEffect, useState } from "react";
// import Modal from "react-modal";
// import ImageModal from "./ImageModal";

// const Card = ({ photo }) => {
//   useEffect(() => {
//     Modal.setAppElement("#root");
//   }, []);
//   const [modalIsOpen, setModalIsOpen] = useState(false);

//   const openModal = () => {
//     setModalIsOpen(true);
//   };

//   const closeModal = () => {
//     setModalIsOpen(false);
//   };
//   return (
//     <div
//       className=" rounded-md p-2 h-48 w-64 shadow-lg
//     bg-opacity-60 backdrop-filter backdrop-blur-lg border border-gray-600
//     "
//     >
//       <img
//         src={photo.img_src}
//         alt={photo.camera.full_name}
//         className="h-32 w-full object-cover cursor-pointer"
//         onClick={openModal}
//       />
//       <div className="mt-2">
//         <h2
//           className="text-sm font-semibold
//         text-white
//         "
//         >
//           {photo.rover.name}
//         </h2>
//         <p
//           className="text-xs
//         text-white
//         "
//         >
//           {photo.earth_date}
//         </p>
//       </div>
//       <ImageModal
//         src={photo.img_src}
//         modalIsOpen={modalIsOpen}
//         closeModal={closeModal}
//         full_name={photo.camera.full_name}
//       />
//     </div>
//   );
// };

// export default Card;

beforeEach(() => {
  const root = document.createElement("div");
  root.id = "root";
  document.body.appendChild(root);
});
describe("Card", () => {
  it("should render Card component", () => {
    const photo = {
      img_src:
        "https://mars.nasa.gov/msl-raw-images/msss/01000/mcam/1000ML0044631200305217E01_DXXX.jpg",
      camera: {
        full_name: "Mast Camera",
      },
      rover: {
        name: "Curiosity",
      },
      earth_date: "2021-08-01",
    };
    const { container } = render(<Card photo={photo} />);
    expect(container).toMatchSnapshot();
  });
});
