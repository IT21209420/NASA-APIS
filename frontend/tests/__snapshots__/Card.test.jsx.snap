// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > should render Card component 1`] = `
<div>
  <div
    class=" rounded-md p-2 h-48 w-64 shadow-lg 
    bg-opacity-60 backdrop-filter backdrop-blur-lg border border-gray-600
    "
  >
    <img
      alt="Mast Camera"
      class="h-32 w-full object-cover cursor-pointer"
      src="https://mars.nasa.gov/msl-raw-images/msss/01000/mcam/1000ML0044631200305217E01_DXXX.jpg"
    />
    <div
      class="mt-2"
    >
      <h2
        class="text-sm font-semibold
        text-white
        "
      >
        Curiosity
      </h2>
      <p
        class="text-xs
        text-white
        "
      >
        2021-08-01
      </p>
    </div>
  </div>
</div>
`;
